### PR TITLE
r.topidx: Rewrite Perl scripts in Python 3

### PR DESCRIPTION
**Pull request by HuidaeCho**
Address #762.

### CODE REVIEW

**Comment by wenzeslaus** [comment]
On a second thought, f-strings are probably okay for these scripts, just applying Black and checking the code with Flake8 should be enough.

**Comment by neteler** [comment]
@HuidaeCho do you plan to make further modifications?

**Comment by HuidaeCho** [comment]
> @HuidaeCho do you plan to make further modifications?

@neteler No, I think this is it. I'm not familiar with Black and Flake8.

**Comment by neteler** [comment]
flake8 reports this:

```
./arc.to.gridatb.py:16:60: E703 statement ends with a semicolon
    print('Usage: arc.to.gridatb.py arc_file gridatb_file');
                                                           ^
```

I'll fix it here.

File: raster/r.topidx/arc.to.gridatb.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    m = re.match(pattern, string, re.IGNORECASE)
+    if m:
+        match.value = m.group(1)
+        return True
+    else:
+        match.value = None
+        return False
+
+if len(sys.argv) != 3 or re.match('^-*help', sys.argv[1]):
+    print('Usage: arc.to.gridatb.py arc_file gridatb_file');
```
Comment (neteler) [inline raster/r.topidx/arc.to.gridatb.py:16]:
```suggestion
    print('Usage: arc.to.gridatb.py arc_file gridatb_file')
```

**Comment by neteler** [comment]
Do you want any other changes @wenzeslaus (since the PR is still in "changes requested" mode)?

**Comment by wenzeslaus** [comment]
> Do you want any other changes @wenzeslaus (since the PR is still in "changes requested" mode)?

I was suggesting only to use Flake8 and Black which seems you did to at least some extent. Fully applying Black would be good.

**Comment by neteler** [comment]
(Please (someone) add the magic black cmd line to

https://trac.osgeo.org/grass/wiki/Submitting/Python )

**Comment by wenzeslaus** [comment]
> Please (someone) add the magic black cmd line to

See _Style > Automation_ at https://trac.osgeo.org/grass/wiki/Submitting/Python#Automation

**Comment by neteler** [comment]
I suggest to merge. The `black` formatting result looks strange to me (tried locally).

**Comment by HuidaeCho** [comment]
> I suggest to merge. The `black` formatting result looks strange to me (tried locally).

@neteler Agreed with you. The `black` result looks unnatural. I just made small changes to make `flake8` happy though.

**Comment by wenzeslaus** [comment]
> > I suggest to merge. The `black` formatting result looks strange to me (tried locally).
> 
> @neteler Agreed with you. The `black` result looks unnatural. I just made small changes to make `flake8` happy though.

The place to have this discussion is #543. Speak now or forever hold your peace.

**Comment by wenzeslaus** [comment]
Since I'm introducing Black elsewhere (#1347) and we have already used some Black here and there, I have applied Black and renamed files to use underscores instead of dots to separate words. Example of what I consider an improvement:

```diff
- if len(sys.argv) == 1 or len(sys.argv) == 4 or len(sys.argv) > 5 or \
-    re.match('^-*help', sys.argv[1]):
-     print('Usage: gridatb.to.arc.py gridatb_file arc_file'
-           ' [xllcorner yllcorner]')
+ if (
+     len(sys.argv) == 1
+     or len(sys.argv) == 4
+     or len(sys.argv) > 5
+     or re.match("^-*help", sys.argv[1])
+ ):
+     print("Usage: gridatb.to.arc.py gridatb_file arc_file [xllcorner yllcorner]")
```

The `(...):` piece seems strange at the beginning, but at the end easier to read than that one space difference in indentation and backslash in the previous code because number of rules applied is smaller. It is worth mentioning that the Black formatting is also meant for version control (subsequent changes to the code should generate small diffs in the sense that changing one `or ...` touches only one line and no other part of the condition).

**Comment by HuidaeCho** [comment]
+1 for merging it whenever it's ready. I don't worry too much about how the code looks.

**Review by wenzeslaus — APPROVED** [review]

**Pull request merged**